src/drivers/Wayland/Fl_Wayland_Screen_Driver.cxx

```diff
@@ -359,6 +359,68 @@ static struct wl_pointer_listener pointer_listener = {
   pointer_axis
 };
 
+static void touch_down(void *data,
+         struct wl_touch *wl_touch,
+         uint32_t serial,
+         uint32_t time,
+         wl_surface *surface,
+         int id,
+         wl_fixed_t x,
+         wl_fixed_t y)
+{
+  struct Fl_Wayland_Screen_Driver::seat *seat = (struct Fl_Wayland_Screen_Driver::seat*)data;
+  seat->serial = serial;
+  wld_event_time = time;
+  Fl_Window *win = event_coords_from_surface(surface, x, y);
+  if (!win) return;
+  win = win->top_window();
```

Comment:
This is superfluous because it's already di$one in event_coords_from_surface() called before.